synapse/storage/room.py

```diff
@@ -171,6 +171,24 @@ def is_room_blocked(self, room_id):
             desc="is_room_blocked",
         )
 
+    @cachedInlineCallbacks(max_entries=10000)
```

Comment:
If we're caching we need to make sure that we invalidate when we change, which we don't seem to here. Given this only happens during invites lets just not bother with caching.

```suggestion
    @defer.inlineCallbacks
```